tests/execution_engine/test_pandas_execution_engine.py

```diff
@@ -41,6 +42,23 @@ def test_constructor():
     assert engine.config.get("boto3_options")["region_name"] == "us-east-1"
 
 
+@mock.patch(
+    "great_expectations.datasource.data_connector.configured_asset_azure_data_connector.BlobServiceClient"
+)
+def test_constructor_with_azure_options(mock_azure_conn):
+    # default instantiation
+    PandasExecutionEngine()
+
+    # instantiation with custom parameters
+    engine = PandasExecutionEngine(discard_subset_failing_expectations=True)
+    assert "discard_subset_failing_expectations" in engine.config
+    assert engine.config.get("discard_subset_failing_expectations") is True
+    custom_azure_options = {"account_url": "my_account_url"}
+    engine = PandasExecutionEngine(azure_options=custom_azure_options)
+    assert "azure_options" in engine.config
+    assert engine.config.get("azure_options")["account_url"] == "my_account_url"
+
+
```

Comment:
Ensured proper instantiation with `azure_options` but that is the extent of tests I've written. The above comment about mocks hindering what we can test holds here. Thoughts on what I should test and how I should test it?

I can mock a lot of the internal calls but at what point do we mock too much?

Reply:
This is a great question, and one that I think we should talk about more (not just in this PR, but as pod / company)... I think the work with BigQuery has revealed some "holes" in our testing that we weren't able to catch precisely because we only test for the internal calls. Would love to bring this up as part of the next PI planning meeting even sooner